pkg/dns/client/dns.go

```diff
@@ -362,13 +363,13 @@ func roundRobin(in []dns.RR) []dns.RR {
 	return out
 }
 
-func roundRobinShuffle(records []dns.RR) {
-	switch l := len(records); l {
+func roundRobinShuffle[T any](entries []T) {
+	switch l := len(entries); l {
 	case 0, 1:
 		break
 	case 2:
 		if dns.Id()%2 == 0 {
```

Comment:
`dns.ID()` simply returns a `crypto/rand`-generated uint16. May be more readable to copy over that function and use directly here?